app-modules/interaction/src/Filament/Actions/DraftInteractionWithAiAction.php

```diff
@@ -0,0 +1,146 @@
+                'avatarUrl' => AiAssistant::query()->where('is_default', true)->first()
+                    ?->getFirstTemporaryUrl(now()->addHour(), 'avatar', 'avatar-height-250px') ?: Vite::asset('resources/images/canyon-ai-headshot.jpg'),
+            ]))
+            ->modalWidth(MaxWidth::ExtraLarge)
+            ->modalSubmitActionLabel('Draft')
+            ->form([
+                Textarea::make('instructions')
+                    ->hiddenLabel()
+                    ->rows(4)
+                    ->placeholder('What do you want to write about?')
+                    ->required(),
+            ])
+            ->action(function (array $data, Get $get, Set $set, Page $livewire) {
+                $service = Feature::active('ai-integrated-assistant-settings')
+                    ? app(AiIntegratedAssistantSettings::class)->default_model->getService()
+                    : app(AiSettings::class)->default_model->getService();
+
+                $userName = auth()->user()->name;
+                $userJobTitle = auth()->user()->job_title ?? 'staff member';
+                $clientName = app(LicenseSettings::class)->data->subscription->clientName;
+                $model = $livewire::getResource()::getModelLabel();
+                $initiative = InteractionInitiative::find($get('interaction_initiative_id'));
+                $driver = InteractionDriver::find($get('interaction_driver_id'));
+                $outcome = InteractionOutcome::find($get('interaction_outcome_id'));
+                $type = InteractionType::find($get('interaction_type_id'));
+
+                try {
+                    $content = $service->complete(
+                        <<<EOL
+                            My name is {$userName}, and I am a {$userJobTitle} at {$clientName}.
+
+                            Please document my interaction with the {$model} {$livewire->record}['full_name'] at our college based on the following details:
+
+                            Instructions:
+                            - Respond only with the interaction content—no greetings or additional comments.
+                            - The first line should be the raw subject of the interaction with no "Subject: " label, written in plain text.
+                            - The interaction body should start on the second line, using plain text only, with no special formatting.
+                            - Never mention in your response that the content is formatted or rendered in plain text.
+                            - Use the following context, if available, to enhance the interaction body (but not the subject line):
+                            1. Mention the related initiative: {$initiative?->name}.
+                            2. Include the call driver: {$driver?->name}.
+                            3. State the interaction outcome: {$outcome?->name}.
+                            4. Specify the type of engagement: {$type?->name}.
```

Comment:
We might want to find a way to only add these to the prompt dynamically if their values are not `null`. As it being `null` and therefore blank in the prompt may very well confuse the AI.

Reply:
> We might want to find a way to only add these to the prompt dynamically if their values are not `null`. As it being `null` and therefore blank in the prompt may very well confuse the AI.

Hi @Orrison ,
I added the dynamic context in the prompt which will only be added if the context has at least one instruction.